### PR TITLE
eve-log: custom tls output and log certificate (v4)

### PR DESCRIPTION
**Pull request by thus**
Add the possibility to specify in suricata.yaml which tls fields to log in eve-log.

```
- tls:
    extended: no
    custom: [subject, issuer, fingerprint, sni, version, not_before, not_after, certificate, chain]
```

Also adds support for logging certificate and full certificate chain in eve-log.

https://redmine.openinfosecfoundation.org/issues/1998

- PR thus-pcap: https://buildbot.openinfosecfoundation.org/builders/thus-pcap/builds/102
- PR thus: https://buildbot.openinfosecfoundation.org/builders/thus/builds/102

### CODE REVIEW

File: src/output-json-tls.c [hunk 1]
```diff
@@ -119,30 +155,145 @@ void JsonTlsLogJSONExtended(json_t *tjs, SSLState * state)
...
+    }
+
+    SSLCertsChain *cert = TAILQ_FIRST(&ssl_state->server_connp.certs);
+    if (cert == NULL) {
+        return;
+    }
+
+    unsigned long len = cert->cert_len * 2;
+    uint8_t encoded[len];
+    Base64Encode(cert->cert_data, cert->cert_len, encoded, &len);
```
Comment (inliniac) [inline src/output-json-tls.c]: Please check retval like in https://github.com/inliniac/suricata/blob/master/src/output-json-alert.c#L399

It's missing in a few other places both in your PR (and in master too). We should probably make Base64Encode with WARN_UNUSED

File: src/util-error.h [hunk 2]
```diff
@@ -323,6 +323,7 @@ typedef enum {
     SC_ERR_DNP3_CONFIG,
     SC_ERR_DIR_OPEN,
     SC_WARN_REMOVE_FILE,
+    SC_WARN_DUPLICATE_OUTPUT,
```
Comment (inliniac) [inline src/util-error.h]: please only append to this enum

**Comment by inliniac** [comment]
Needs a bit of work.

**Comment by thus** [comment]
Replaced by #2580

**Pull request closed without merging**